src/components/elements/Map-mapbox/MapControls/PolygonModifier.tsx

```diff
@@ -0,0 +1,100 @@
+import { useT } from "@transifex/react";
+import { useState } from "react";
+
+import IconButton from "@/components/elements/IconButton/IconButton";
+import ControlButtonsGroup from "@/components/elements/Map-mapbox/components/ControlButtonsGroup";
+import ControlDivider from "@/components/elements/Map-mapbox/components/ControlDivider";
+import { IconNames } from "@/components/extensive/Icon/Icon";
+import ModalConfirm from "@/components/extensive/Modal/ModalConfirm";
+import { useModalContext } from "@/context/modal.provider";
+import { useSitePolygonData } from "@/context/sitePolygon.provider";
+import { fetchDeleteV2TerrafundProjectPolygonUuid } from "@/generated/apiComponents";
+
+import Button from "../../Button/Button";
+import Text from "../../Text/Text";
+
+interface PolygonModifierProps {
+  polygonFromMap: { uuid: string; isOpen: boolean } | undefined;
+  onClick?: () => void;
+  onSave?: () => void;
+  onCancel?: () => void;
+}
+
+const PolygonModifier = ({ polygonFromMap, onClick, onSave, onCancel }: PolygonModifierProps) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const { openModal, closeModal } = useModalContext();
+  const contextSite = useSitePolygonData();
+  const reloadSiteData = contextSite?.reloadSiteData;
+  const t = useT();
+
+  const handleSave = () => {
+    onSave?.();
+    setIsEditing(false);
+    reloadSiteData?.();
+  };
+
+  const handleDelete = async () => {
+    if (polygonFromMap?.uuid) {
+      await fetchDeleteV2TerrafundProjectPolygonUuid({ pathParams: { uuid: polygonFromMap.uuid } });
+      reloadSiteData?.();
+    }
+  };
+
+  const openDeleteConfirmation = () => {
+    openModal(
+      <ModalConfirm
+        title={t("Confirm Polygon Deletion")}
+        content={t("Do you want to delete this polygon?")}
+        onClose={closeModal}
+        onConfirm={() => {
+          handleDelete();
+          closeModal();
+        }}
+      />
+    );
+  };
+
+  return (
+    <ControlButtonsGroup direction="col" className="z-10 w-auto">
+      <IconButton
+        iconProps={{ name: IconNames.EDIT, width: 24, height: 24 }}
+        onClick={() => {
+          setIsEditing(true);
+          onClick?.();
+        }}
+        className="rounded-b-none rounded-t-lg p-[10px]"
+        aria-label="Edit Polygon"
```

Comment:
Make sure to translate text. I see this and the `Save` and `Cancel` text below at least that need an update.

Reply:
done

Reply:
Looks like this file still needs doing.

Reply:
Sorry, forget that one. 
Already checked again.